exercises/diffie-hellman/canonical-data.json

```diff
@@ -0,0 +1,67 @@
+{
+  "exercise": "diffie-hellman",
+  "version": "1.0.0",
+  "comments": [
+    "Optional tests to consider:",
+    "* Validation of parameters:",
+    "  Validate that `p`, `g` are valid.",
+    "  Validate that keys given as inputs are valid.",
+    "  Resources that show what happens if parameters are not validated:",
+    "  http://cryptopals.com/sets/5/challenges/34",
+    "  http://cryptopals.com/sets/5/challenges/35",
+    "* Large numbers:",
+    "  Although the calculations fundamentally do not require large numbers,",
+    "  this is a reasonable real-world use for them",
+    "  and it may be instructive to have an exercise on their use.",
+    "  Consult tracks with this exercise (such as the Go track) for possible inputs to use."
+  ],
+  "cases": [
+    {
+      "description": "private key is in range 1 .. p",
+      "property": "privateKeyIsInRange",
+      "input": {},
+      "expected": {}
```

Comment:
I'm not here to tell individual tracks how they test this, so I have no value for `expected` to put here. Though I guess I can throw out the following alternative proposals:

* `"expected": {}` (what I currently have)
* `"expected": "1 < privateKey < p` (expressing it as a string)
* `"expected": {"greaterThan": 1, "lessThan": "p"} (expressing it in a more structured way)
* Your suggestion here.

Reply:
> - `"expected": {"greaterThan": 1, "lessThan": "p"}`

Could we indicate that this test calls `property` multiple times, verifying all returned values are in range?

Reply:
> > - `"expected": {"greaterThan": 1, "lessThan": "p"}`
>
> Could we indicate that this test calls `property` multiple times, verifying all returned values are in range?

While I don't want to force tracks to implement their tests in any particular way, I can be convinced that it is universally wise (across all tracks) to generate multiple private keys, since even an incorrect implementation may, by chance, generate a single correct private key and therefore not be found to be incorrect by a test that generated only once.

Conclusion: I'm happy to use any means suggested to indicate this, since there is no precedent.

Reply:
I would be in favor of being as explicit as possible, meaning I prefer `"expected": {"greaterThan": 1, "lessThan": "p"}`.